### ISSUE TITLE
Update 12.0.2 -> 12.0.3 failing 

### ISSUE THREAD

**Issue by MegaV0lt** [nextcloud/updater#142]
Update is failing at "backup"

    Initializing
    Current version is 12.0.2.
    Update to Nextcloud 12.0.3 available. (channel: "stable")
    Following file will be downloaded automatically: https://download.nextcloud.com/server/releases/nextcloud-12.0.3.zip
    Check for expected files
    Check for write permissions
    Enable maintenance mode
    Create backup
    Parsing response failed. <!DOCTYPE html> <html class="ng-csp" data-placeholder-focus="false" lang="de" > <head data-requesttoken="WpWqN1NO+UpG6tcmPfeUN0IsOk9jF3RqWuc5A7oLJS4=:IKPTbTQjticPuLUXaZX/ZBRofBZWex8EDNVfc4w5E2A="> <meta charset="utf-8"> <title> Nextcloud </title> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <meta name="referrer" content="never"> <meta name="viewport" content="width=device-width, minimum-scale=1.0, maximum-scale=1.0"> <meta name="apple-itunes-app" content="app-id=1125420102"> <meta name="theme-color" content="#0082c9"> <link rel="icon" href="/nextcloud/core/img/favicon.ico"> <link rel="apple-touch-icon-precomposed" href="/nextcloud/core/img/favicon-touch.png"> <link rel="mask-icon" sizes="any" href="/nextcloud/core/img/favicon-mask.svg" color="#0082c9"> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" type="text/javascript"> var oc_debug=false; var oc_isadmin=true; var oc_dataURL="data"; var oc_webroot="/nextcloud"; var oc_appswebroots={"activity":"/nextcloud/apps/activity","audioplayer":"/nextcloud/apps/audioplayer","bruteforcesettings":"/nextcloud/apps/bruteforcesettings","calendar":"/nextcloud/apps/calendar","comments":"/nextcloud/apps/comments","contacts":"/nextcloud/apps/contacts","dav":"/nextcloud/apps/dav","federatedfilesharing":"/nextcloud/apps/federatedfilesharing","federation":"/nextcloud/apps/federation","files":"/nextcloud/apps/files","files_pdfviewer":"/nextcloud/apps/files_pdfviewer","files_sharing":"/nextcloud/apps/files_sharing","files_texteditor":"/nextcloud/apps/files_texteditor","files_trashbin":"/nextcloud/apps/files_trashbin","files_versions":"/nextcloud/apps/files_versions","files_videoplayer":"/nextcloud/apps/files_videoplayer","firstrunwizard":"/nextcloud/apps/firstrunwizard","gallery":"/nextcloud/apps/gallery","logreader":"/nextcloud/apps/logreader","lookup_server_connector":"/nextcloud/apps/lookup_server_connector","nextcloud_announcements":"/nextcloud/apps/nextcloud_announcements","notifications":"/nextcloud/apps/notifications","oauth2":"/nextcloud/apps/oauth2","password_policy":"/nextcloud/apps/password_policy","provisioning_api":"/nextcloud/apps/provisioning_api","serverinfo":"/nextcloud/apps/serverinfo","sharebymail":"/nextcloud/apps/sharebymail","survey_client":"/nextcloud/apps/survey_client","systemtags":"/nextcloud/apps/systemtags","templateeditor":false,"theming":"/nextcloud/apps/theming","twofactor_backupcodes":"/nextcloud/apps/twofactor_backupcodes","updatenotification":"/nextcloud/apps/updatenotification","workflowengine":"/nextcloud/apps/workflowengine"}; var datepickerFormatDate="dd.MM.yy"; var nc_lastLogin=0; var dayNames=["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"]; var dayNamesShort=["Sun.","Mon.","Tue.","Wed.","Thu.","Fri.","Sat."]; var dayNamesMin=["Su","Mo","Tu","We","Th","Fr","Sa"]; var monthNames=["January","February","March","April","May","June","July","August","September","October","November","December"]; var monthNamesShort=["Jan.","Feb.","Mar.","Apr.","May.","Jun.","Jul.","Aug.","Sep.","Oct.","Nov.","Dec."]; var firstDay=1; var oc_config={"session_lifetime":1440,"session_keepalive":true,"version":"12.0.2.0","versionstring":"12.0.2","enable_avatars":true,"lost_password_link":null,"modRewriteWorking":false,"sharing.maxAutocompleteResults":0,"sharing.minSearchStringLength":0,"blacklist_files_regex":"\\.(part|filepart)$"}; var oc_appconfig={"core":{"defaultExpireDateEnabled":false,"defaultExpireDate":null,"defaultExpireDateEnforced":null,"enforcePasswordForPublicLink":false,"enableLinkPasswordByDefault":false,"sharingDisabledForUser":false,"resharingAllowed":true,"remoteShareAllowed":true,"federatedCloudShareDoc":"https:\/\/docs.nextcloud.com\/server\/12\/go.php?to=user-sharing-federated","allowGroupSharing":true},"shareByMailEnabled":true,"shareByMail":{"enforcePasswordProtection":false}}; var oc_defaults={"entity":"Nextcloud","name":"Nextcloud","title":"Nextcloud","baseUrl":"https:\/\/nextcloud.com","syncClientUrl":"https:\/\/nextcloud.com\/install\/#install-clients","docBaseUrl":"https:\/\/docs.nextcloud.com","docPlaceholderUrl":"https:\/\/docs.nextcloud.com\/server\/12\/go.php?to=PLACEHOLDER","slogan":"ein sicherer Ort f\u00fcr all Deine Daten","logoClaim":"","shortFooter":"<a href=\"https:\/\/nextcloud.com\" target=\"_blank\" rel=\"noreferrer\">Nextcloud<\/a> \u2013 ein sicherer Ort f\u00fcr all Deine Daten","longFooter":"<a href=\"https:\/\/nextcloud.com\" target=\"_blank\" rel=\"noreferrer\">Nextcloud<\/a> \u2013 ein sicherer Ort f\u00fcr all Deine Daten","folder":""}; var oc_userconfig={"avatar":{"version":0}}; </script> <link rel="stylesheet" href="/nextcloud/core/vendor/select2/select2.css?v=dc597945ad7e1e9ce157ec094d9b1172-0"> <link rel="stylesheet" href="/nextcloud/core/vendor/jquery-ui/themes/base/jquery-ui.css?v=dc597945ad7e1e9ce157ec094d9b1172-0"> <link rel="stylesheet" href="/nextcloud/core/css/jquery-ui-fixes.css?v=dc597945ad7e1e9ce157ec094d9b1172-0"> <link rel="stylesheet" href="/nextcloud/core/css/jquery.ocdialog.css?v=dc597945ad7e1e9ce157ec094d9b1172-0"> <link rel="stylesheet" href="/nextcloud/apps/files_versions/css/versions.css?v=dc597945ad7e1e9ce157ec094d9b1172-0"> <link rel="stylesheet" href="/nextcloud/core/css/guest.css?v=dc597945ad7e1e9ce157ec094d9b1172-0"> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/core/vendor/core.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/index.php/js/core/merged-template-prepend.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/core/search/js/search.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/core/l10n/de.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/core/js/jquery-ui-fixes.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/core/js/files/fileinfo.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/core/js/files/client.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> <script nonce="V3BXcU4xTk8rVXBHNnRjbVBmZVVOMElzT2s5akYzUnFXdWM1QTdvTEpTND06SUtQVGJUUWp0aWNQdUxVWGFaWC9aQlJvZkJaV2V4OEVETlZmYzR3NUUyQT0=" src="/nextcloud/core/js/contactsmenu.js?v=dc597945ad7e1e9ce157ec094d9b1172-0"></script> </head> <body id="body-login"> <noscript> <div id="nojavascript"> <div> Diese Anwendung benötigt JavaScript zum ordnungsgemäßen Betrieb. Bitte <a href="http://enable-javascript.com/" target="_blank" rel="noreferrer">aktiviere JavaScript</a> und lade die Seite neu. </div> </div> </noscript> <div class="wrapper"> <div class="v-align"> <header role="banner"> <div id="header"> <div class="logo"> <h1 class="hidden-visually"> Nextcloud </h1> </div> <div id="logo-claim" style="display:none;"></div> </div> </header> <ul> <li class='error'> Zugriff verboten<br> <p class='hint'></p> </li> </ul> <div class="push"></div> </div> </div> <footer role="contentinfo"> <p class="info"> <a href="https://nextcloud.com" target="_blank" rel="noreferrer">Nextcloud</a> – ein sicherer Ort für all Deine Daten </p> </footer> </body> </html>

Running on Debian 9 64 Bit.

```
Config:
    [~] # cat /var/www/nextcloud/config/config.php 
    <?php
    $CONFIG = array (
      'instanceid' => 'xxx',
      'passwordsalt' => 'xxx',
      'secret' => 'xxx',
      'trusted_domains' => 
      array (
        0 => 'xxx',
        1 => 'HP-T5730',
      ),
      'datadirectory' => '/var/www/nextcloud/data',
      'overwrite.cli.url' => 'https://xxx/nextcloud',
      'dbtype' => 'mysql',
      'version' => '12.0.2.0',
      'dbname' => 'nextcloud',
      'dbhost' => 'localhost',
      'dbport' => '',
      'dbtableprefix' => 'oc_',
      'dbuser' => 'oc_Darkwing3',
      'dbpassword' => 'xxx',
      'memcache.local' => '\\OC\\Memcache\\APCu',
      'logtimezone' => 'UTC',
      'installed' => true,
      'updater.release.channel' => 'stable',
      'mail_smtpmode' => 'smtp',
      'mail_from_address' => 'xxx',
      'mail_domain' => 'xxx',
      'mail_smtpauth' => 1,
      'mail_smtphost' => 'mail.gmx.net',
      'mail_smtpsecure' => 'tls',
      'mail_smtpname' => 'xxx,
      'mail_smtppassword' => 'xxx',
      'maintenance' => true,
      'theme' => '',
      'loglevel' => 2,
      'updater.secret' => 'xxx',
    );
```
as postet in ncforum: http://j.mp/2fBchJU

**Comment by LukasReschke** [comment]
Judging from the response the webserver returns a 403 on the backup step:

```
<h1 class="hidden-visually"> Nextcloud </h1> </div> <div id="logo-claim" style="display:none;"></div> </div> </header> <ul> <li class='error'> Zugriff verboten<br>
```

Are you running mod_security or any such tools to block requests? Also, can you check your web server error logs for the appearance and reason of this 403?

**Comment by MegaV0lt** [comment]
in error.log is this:
`[Thu Sep 21 08:25:12.512551 2017] [:error] [pid 28529] [client 24.137.76.79:46843] script '/var/www/html/index.php' not found or unable to stat
[Thu Sep 21 08:50:13.597908 2017] [authz_core:error] [pid 2803] [client 87.176.227.187:40718] AH01630: client denied by server configuration: /var/www/nextcloud/data/.ocdata
[Thu Sep 21 08:50:23.714726 2017] [evasive20:error] [pid 2803] [client 87.176.227.187:40718] client denied by server configuration: /var/www/nextcloud/updater/index.php`

Any hint how io can fix this?

**Comment by T7o7heVV** [comment]
Did the same update 12.0.2 -> 12.0.3 and now it's dead. Stuck at "maintenance mode" message when trying to login. :(

**Comment by psylock** [comment]
T7o7heVV +1.
 
Where can i see updating log ?

**Comment by MegaV0lt** [comment]
In my case the problem was a changed hostname. After putting it in "trusted_domains" and setting maintenance to false the update was made without problems.

**Comment by psylock** [comment]
thanks !

i put the maintenance mode to false, and follow the instructions to update with ./occ upgrade, it worked.

**Comment by T7o7heVV** [comment]
I did not look at the log, but simply saw that the website was stuck at
"maintenance".

I just solved it by setting the maintenance mode to 'false'. I looked at
this documentation:
https://docs.nextcloud.com/server/9/admin_manual/maintenance/enable_maintenance.html

After I restarted the nextcloud docker-container, the webportal showed my
the "start update" (translated form Germany, don't know actual english
phrase) button. As it told me, I made a backup of the DB (just in case) and
clicked the button.

It started doing an update and then said "an error occurred". I reloaded
the webpage and strangely enough, every seems to be working now. Well I
guess only time will tell.


2017-09-27 14:47 GMT+02:00 François-Xavier Guillois <
notifications@github.com>:
